src/assets/js/search.js

```diff
@@ -0,0 +1,74 @@
+(() => {
+  const getExcerpt = (content, query) => {
+    const queryRegex = new RegExp(query, "i");
+    const searchIndex = content.search(queryRegex);
+    const queryLength = query.length;
+    const excerpt = content.slice(
+      searchIndex - 100,
+      searchIndex + queryLength + 100
+    );
+
+    return excerpt.replace(
+      queryRegex,
+      "<strong class='search-keyword'>$&</strong>"
+    );
+  };
+
+  const displaySearchResults = (results, store, searchTerm) => {
+    const searchResults = document.getElementById("search-results");
+
+    if (results.length) {
+      let appendString = "";
+
+      for (let i = 0; i < results.length; i++) {
+        const item = store[results[i].ref];
+        appendString +=
+          '<li><a href="' + item.url + '"><h3>' + item.title + "</h3></a>";
+        appendString +=
+          "<p>..." + getExcerpt(item.content, searchTerm) + "...</p></li>";
+      }
+
+      searchResults.innerHTML = appendString;
+    } else {
+      searchResults.innerHTML = "<li>No results found</li>";
+    }
+  };
+
+  const getQueryVariable = (variable) => {
+    const query = window.location.search.substring(1);
+    const vars = query.split("&");
```

Comment:
When would there be an ampersand in the query string?

Reply:
I think this was lazily copied and pasted from that tutorial, your suggestion is a lot cleaner. The one thing maybe worth thinking about is if we add multiple search terms, they're separated by `+` in the search bar, so there may be a reason we'd want to split them and search for them independently, but I think that's a decision we can defer until later.